### PR TITLE
mgmt, update codegen to 4.0.37

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Also update script to avoid deleting hand-written samples.

### CODE REVIEW

File: eng/mgmt/automation/generate.py [hunk 1]
```diff
@@ -48,7 +48,11 @@ def generate(
         module,
     )
     shutil.rmtree(os.path.join(output_dir, 'src/main'), ignore_errors = True)
-    shutil.rmtree(os.path.join(output_dir, 'src/samples'), ignore_errors = True)
+    if os.path.exists(os.path.join(output_dir, 'src/samples/README.md')):
+        # samples contains hand-written code
+        shutil.rmtree(os.path.join(output_dir, 'src/samples/java', namespace.replace('.', '/'), 'generated'), ignore_errors = True)
+    else:
+        shutil.rmtree(os.path.join(output_dir, 'src/samples'), ignore_errors = True)
```
Comment (haolingdong-msft) [inline eng/mgmt/automation/generate.py:55]: If no hand-written sample, the sample folder path will not contain /generated?

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/generate.py:55]: If no hand-written sample, we just delete the `samples` folder, delete everything.

Comment (haolingdong-msft) [reply eng/mgmt/automation/generate.py:55]: If we:
1. generate sample under samples/
2. add manual samples
will the generated sample be updated in the future ?

**Review by haolingdong-msft — APPROVED** [review]

File: eng/mgmt/automation/generate.py (hunk 1 above)
Comment (weidongxu-microsoft) [reply eng/mgmt/automation/generate.py:55]: Firstly we would almost never add manual sample to Lite SDK.

Secondly, if we do add manual sample and see that folder already have generated samples, the best approach maybe doing a regen first, then add manual sample. Or one can move them manually. Any case, this probably not going to happen.

Comment (haolingdong-msft) [reply eng/mgmt/automation/generate.py:55]: ok, just want to make sure manual sample will not be deleted.

**Pull request merged**